pom.xml

```diff
@@ -185,6 +203,31 @@
                         </execution>
                     </executions>
                 </plugin>
+                <plugin>
+                    <groupId>com.diffplug.spotless</groupId>
+                    <artifactId>spotless-maven-plugin</artifactId>
+                    <version>${maven.plugin.spotless.version}</version>
+                    <configuration>
+                        <upToDateChecking>
+                            <enabled>true</enabled>
+                        </upToDateChecking>
+                        <python>
+                            <includes>
+                                <include>${spotless.python.includes}</include>
+                            </includes>
+                            <black>
+                                <version>${spotless.python.black.version}</version>
+                            </black>
```

Comment:
The key changes are applied here.
1. `spotless.python.includes` for included path remains empty by default, unless `spotless-python` profile is enabled. This is intended to avoid errors when running Maven builds without `black` installed.

2. pinning black version is essential for spotless maven plugin's Python styling

cc @lexluo09